samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,219 @@
+
+This section explains operations and corresponding mesasge sequence.
+
+* Retrieve current status of the device:
+
+   1. The configurator prepares ``Request`` message with ``GET_STATUS`` command and sends encoded message to the device.
+   #. The device decoded the message and finds the opcode is ``GET_STATUS``.
+   #. The device retrieves current device status information.
+   #. The device prepares ``Response`` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+
+* Start scan and find out access point near the device:
+
+   1. The configurator prepares ``Request`` message with ``START_SCAN`` command and sends encoded message to the device.
+   #. The device decoded the message and finds the opcode is ``START_SCAN``.
+   #. The device sends scan request to Wi-Fi subsystem. If scan parameter is provided in the message, it is sent together.
+   #. The device prepares ``Response`` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+   #. When the scan is done, the device prepares ``Result`` message and sends encoded message to the configurator.
+
+* Stop scanning access point:
+
+   1. The configurator prepares ``Request`` message with ``STOP_SCAN`` command and sends encoded message to the device.
+   #. The device decoded the message and finds the opcode is ``STOP_SCAN``.
+   #. The device sends stop scan request to Wi-Fi subsystem.
+   #. The device prepares ``Response`` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+
+* Pass Wi-Fi credential and let the device apply it:
+
+   1. The configurator prepares ``Request`` message with ``SET_CONFIG`` command and sends encoded message to the device.
+   #. The device decoded the message and finds the opcode is ``SET_CONFIG``.
+   #. The device sends connection request to Wi-Fi subsystem. The connection parameter is sent as well.
+   #. The device prepares ``Response`` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+   #. When the connection is done, the device prepares ``Result`` message and sends encoded message to the configurator. The Wi-Fi credential will be saved in flash if connection is successful.
+
+* Remove the Wi-Fi credential in the device and disconnect if it is connected currently:
+
+   1. The configurator prepares ``Request`` message with ``FORGET_CONFIG`` command and sends encoded message to the device.
+   #. The device decoded the message and finds the opcode is ``FORGET_CONFIG``.
+   #. The device sends disconnection request to Wi-Fi subsystem, and removes saved Wi-Fi credential if any.
+   #. The device prepares ``Response`` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+
```

Comment:
```suggestion
```

As discussed in chat, the parameters for specific requests are defined above, so we can take this command-specific information out.